src/backend/services/workflow_service.py

```
@@ -89,17 +89,40 @@ def run_agentic_workflow(natural_language_query: str, model_provider: str, model
 
-        # Extract robot code from task[2] (code_assembler - no more popup task)
-        robot_code = crew_with_results.tasks[2].output.raw
+        # Extract robot code from task[2] (code_assembler)
+        # With output_pydantic=AssemblyOutput, code is in output.pydantic.code
+        # Fall back to output.raw for backward compatibility
+        task_output = crew_with_results.tasks[2].output
+        
+        # Strategy 1: Try Pydantic output (new format with output_pydantic)
+        if hasattr(task_output, 'pydantic') and task_output.pydantic:
+            robot_code = task_output.pydantic.code
+            logging.info("✅ Extracted robot code from output.pydantic.code (AssemblyOutput)")
```

Ruff (0.14.8)

`info()` call on root logger

(LOG015)